pkg/pgmodel/cache/series_cache.go

```diff
@@ -10,31 +10,92 @@ import (
 	"math"
 	"sort"
 	"strings"
+	"time"
 
 	"github.com/prometheus/prometheus/pkg/labels"
 	"github.com/timescale/promscale/pkg/clockcache"
+	"github.com/timescale/promscale/pkg/log"
 	"github.com/timescale/promscale/pkg/pgmodel/model"
 	"github.com/timescale/promscale/pkg/prompb"
 )
 
-//this seems like a good upper bound on default /active/ series. Takes about 64MB
-const DefaultSeriesCacheSize = 500000
+//this seems like a good initial size for /active/ series. Takes about 32MB
+const DefaultSeriesCacheSize = 250000
+
+const GrowCheckDuration = time.Minute //check whether to grow the series cache this often
+const GrowEvictionThreshold = 0.2     // grow when evictions more than 20% of cache size
+const GrowFactor = float64(2.0)       // multiply cache size by this factor when growing the cache
 
 type SeriesCache interface {
 	Reset()
 	GetSeriesFromProtos(labelPairs []prompb.Label) (series *model.Series, metricName string, err error)
 	Len() int
 	Cap() int
+	Evictions() uint64
 }
 
 type SeriesCacheImpl struct {
-	cache *clockcache.Cache
+	cache        *clockcache.Cache
+	maxSizeBytes uint64
+}
+
+func NewSeriesCache(config Config, closer <-chan struct{}) *SeriesCacheImpl {
```

Comment:
Just as a growing series cache is important, isn't a growing label cache important (which is again a clockcache)? This takes me to next suggestion which is, we should have evictions rate checking and growth in the clockcache, running in a background goroutine.

PS: this looks like Prom's ewma rate.

Reply:
I don't think this should be part of clockcache, that code is complicated enough and should probably be just the plain datastructure (it's extremely performance critical). We may write a wrapper for clockcache that does this, but let's do in another PR. Also we can do the labels cache in another PR. This is complicated as-is.

Reply:
We've never seen the labels cache get too full, so we can wait for another PR